skada/metrics.py

```diff
@@ -619,3 +619,125 @@ def _score(self, estimator, X, y, sample_domain=None):
+        Random number generator.
+    _sign : int
+        1 if greater_is_better is True, -1 otherwise.
+
+    References
+    ----------
+    .. [32] Dapeng Hu et al. Mixed Samples as Probes for Unsupervised Model
+            Selection in Domain Adaptation.
+            NeurIPS, 2023.
+    """
+
+    def __init__(
+        self,
+        lmbd=0.55,
+        random_state=None,
+        greater_is_better=True,
+    ):
+        super().__init__()
+        self.lmbd = lmbd
+        self.random_state = random_state
+        self._sign = 1 if greater_is_better else -1
+
+    def _generate_mixed_samples(self, X, y_pred):
+        """
+        Generate mixed samples using mixup.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The input samples.
+        y_pred : array-like of shape (n_samples, n_classes)
+            The predicted probabilities for each class.
+
+        Returns
+        -------
+        X_mixed : array-like of shape (n_samples, n_features)
+            The mixed input samples.
+        y_mixed : array-like of shape (n_samples, n_classes)
+            The mixed labels.
+        """
+        rng = check_random_state(self.random_state)
+        n_samples, n_features = X.shape
+
+        # Generate indices for mixing
+        idx = np.arange(n_samples)
+        rand_idx = rng.randint(0, n_samples, size=n_samples)
+
+        # Mix samples
+        X_mixed = self.lmbd * X[idx] + (1 - self.lmbd) * X[rand_idx]
+        y_mixed = self.lmbd * y_pred[idx] + (1 - self.lmbd) * y_pred[rand_idx]
+
+        return X_mixed, y_mixed
+
+    def _score(self, estimator, X, y=None, sample_domain=None, **params):
+        """
+        Compute the Interpolation Consistency Evaluation (ICE) score.
+
+        Parameters
+        ----------
+        estimator : object
+            The fitted estimator to evaluate.
+        X : array-like of shape (n_samples, n_features)
+            The input samples.
+        y : Ignored
+            Not used, present for API consistency by convention.
+        sample_domain : array-like, default=None
+            Domain labels for each sample.
+
+        Returns
+        -------
+        score : float
+            The ICE score.
+        """
+        X, _, sample_domain = check_X_y_domain(X, y, sample_domain)
+        source_idx = extract_source_indices(sample_domain)
+
+        # Get predictions for target samples
+        y_pred = estimator.predict_proba(
+            X[~source_idx], sample_domain=sample_domain[~source_idx]
+        )
+
+        # Generate mixed samples
+        X_mixed, y_mixed = self._generate_mixed_samples(X[~source_idx], y_pred)
+
+        # Get predictions for mixed samples
+        y_pred_mixed = estimator.predict_proba(
+            X_mixed, sample_domain=np.full(X_mixed.shape[0], -1)
+        )
+
+        # Calculate ICE score
+        ice_score = accuracy_score(y_mixed.argmax(axis=1), y_pred_mixed.argmax(axis=1))
```

Comment:
In the paper, they don't compute accuracy on all the mixed samples. They separate samples with the same `y_pred` and a different one. Then, they compute the accuracy score for diff_mixed_samples and same_mixed_sample and take the mean of the two scores. See Appendix A of the paper.